core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexTemplateService.java

```diff
@@ -194,12 +195,16 @@ private static void validateAndAddTemplate(final PutRequest request, IndexTempla
         Index createdIndex = null;
         final String temporaryIndexName = UUIDs.randomBase64UUID();
         try {
+            // use the provided values, otherwise just pick valid dummy values
+            int dummyPartitionSize = request.settings.getAsInt(IndexMetaData.SETTING_PARTITION_SIZE, 1);
+            int dummyShards = request.settings.getAsInt(IndexMetaData.SETTING_NUMBER_OF_SHARDS,
+                    dummyPartitionSize == 1 ? 1 : dummyPartitionSize + 1);
 
             //create index service for parsing and validating "mappings"
             Settings dummySettings = Settings.builder()
                 .put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT)
                 .put(request.settings)
-                .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, 1)
+                .put(IndexMetaData.SETTING_NUMBER_OF_SHARDS, dummyShards)
```

Comment:
Right, it didn't seem like it being 1 was particularly important but we do need to set a number of shards in situations where _only_ the `partition_size` is provided.

For example, someone could create a template that omits `number_of_shards` but includes a `partition_size` of 10 and this dummy index would default to having a number of shards `number_of_shards` of 5, which fails validation.

I can remove the special handling to default it to 1 by just changing the dummyShards value to `request.settings.getAsInt(IndexMetaData.SETTING_NUMBER_OF_SHARDS, dummyPartitionSize + 1)`